### ISSUE TITLE
Enable serialization output / input stream customization

### ISSUE THREAD

**Issue by timklge** [vaadin/kubernetes-kit#168]
We have recently considered using Kubernetes Kit for session replication in our project. To minimize changes to our Kotlin Spring Boot codebase, we would like to customize the `TransientInjectableObjectOutputStream` to exclude all referenced Spring beans from serialization without needing to annotate them with `@Transient`.

Currently, Kubernetes Kit does not support this because the `SessionSerializer` directly instantiates the output stream responsible for inspecting transient fields within a private method, preventing us from injecting a custom implementation.

We propose introducing a `SerializationStreamFactory` bean to obtain output and input stream instances, allowing us to inject our custom implementation. This custom implementation will replace all Spring beans in `replaceObject`. In our prototype using a forked version of Kubernetes Kit, this approach works well.

If you are open to this change, I will open a PR with the proposed modifications:
[feature/serialization-stream-factory](https://github.com/vaadin/kubernetes-kit/compare/main...timklge:kubernetes-kit:feature/serialization-stream-factory)

**Comment by mcollovati** [comment]
May I ask you why a custom implementation of  `TransientHandler` is not enough for your needs?

**Comment by mcollovati** [comment]
I see.
From a Vaadin point of view, lambda referenced by serializable components should be serializable themselves.
For example, if the lambda is defined in a UI component, usually you can inject the bean in the component and store it in a transient field and reference that field in the lambda.
See https://vaadin.com/docs/latest/tools/kubernetes/session-replication#unserializable-objects-in-lambdas for an example,

Anyway, perhaps there could be other good use cases for a custom TransientInjectableObjectOutputStream.

**Comment by jorgheymans** [comment]
The compression part could be interesting for us, as some of our pages add up to 4MB of data being serialized to the session. Is the compression overhead measurable or something not to worry about at all ?

**Comment by jorgheymans** [comment]
@mcollovati is this being considered in one way or another ?

**Comment by mcollovati** [comment]
> [@mcollovati](https://github.com/mcollovati) is this being considered in one way or another ?

Let's ask @heruan

**Comment by tamasmak** [comment]
> is this being considered in one way or another ?
@jorgheymans Yes, this has been considered, just there are some other works with the kit which have postponed this. We're going handle this as soon as possible.

**Comment by jorgheymans** [comment]
Meantime, I looked a bit deeper into the parts that are of our interest and at least for Redis, the Lettuce driver has a way to handle compression using a Codec: https://redis.github.io/lettuce/integration-extension/#compression . So, for us this could be a solution as well. 

Exposing metrics on the serialization mechanism is also very useful to us, so that part may still need this customization.
